### PR TITLE
31 create movie detail page

### PR DESCRIPTION
**Pull request by systm-spec**
Ready for review

### CODE REVIEW

File: src/api/movies.ts [hunk 1]
```diff
@@ -11,9 +11,7 @@ export const getCurrentMovie = async ({
 }) => {
   const { movieId } = params;
   const res = await axios.get(
-    `https://api.themoviedb.org/3/movie/${movieId}?api_key=${
-      import.meta.env.VITE_TMDB_KEY
-    }`
+    `https://api.themoviedb.org/3/movie/${movieId}?api_key=830580354c9aaf7a57d3f02a7a0010ae`
```
Comment (ReallyBoard123) [inline src/api/movies.ts:14]: never commit your apikey. Please store the API key in the .env file and use the readme from the project to see how to use it :)

File: src/pages/MovieDetails.tsx [hunk 2]
```diff
@@ -1,9 +1,123 @@
...
+  const [novel, setNovel] = useState('');
+  const {
+    id,
+    title,
+    backdrop_path,
+    runtime,
+    vote_average,
+    release_date,
+    overview,
+    genres,
```
Comment (ReallyBoard123) [inline src/pages/MovieDetails.tsx:23]: genres have a type error here. Are you trying to use genre_ids?

File: src/pages/MovieDetails.tsx [hunk 3]
```diff
@@ -1,9 +1,123 @@
...
+  } = movie;
+
+  const getData = () => {
+    axios
+      .get<Credits[]>(
+        `https://api.themoviedb.org/3/movie/${id}/credits?api_key=830580354c9aaf7a57d3f02a7a0010ae`
+      )
+      .then(res => {
+        setDirector(res.data.crew.find(x => x.job == 'Director').name);
+        setNovel(res.data.crew.find(x => x.job == 'Novel').name);
```
Comment (ReallyBoard123) [inline src/pages/MovieDetails.tsx:33]: ummm, I think this has to be little more thoughtful? So the current problem is that the API is a little bit inconsistent. For many movies, it does have a writer - but sometimes available in the known_for_department == Writing. Then you can at least return one writer from there for sure!

**Review by ReallyBoard123 — CHANGES_REQUESTED** [review]
I did not really look deeply into the tailwind styling yet, but the page overall looks great! There are some type errors I think. Sometimes the errors are only available in the console. Thus, having an extension that brings all the type errors in your editor itself helps!

File: src/api/movies.ts (hunk 1 above)
Comment (systm-spec) [reply src/api/movies.ts:14]: Done

File: src/pages/MovieDetails.tsx (hunk 2 above)
Comment (systm-spec) [reply src/pages/MovieDetails.tsx:23]: Reworked

File: src/pages/MovieDetails.tsx (hunk 3 above)
Comment (systm-spec) [reply src/pages/MovieDetails.tsx:33]: cleaned up, added extra case

**Comment by systm-spec** [comment]
Ready for next review

**Review by ReallyBoard123 — CHANGES_REQUESTED** [review]
hey, somehow some of the layout issues persist :( I think some names are too long! and that breaks the page. I am sorry, but I did not think of this before. So names would have to be clipped! You would have to write a JS function for this.

try this link: http://localhost:5173/movies/569094
another one: http://localhost:5173/movies/575264

and the writer seems to be still missing here: http://localhost:5173/movies/667538

even though this is there in the response
{
      "adult": false,
      "gender": 2,
      "id": 74569,
      "known_for_department": "Writing",
      "name": "Joby Harold",
      "original_name": "Joby Harold",
      "popularity": 8.216,
      "profile_path": "/q3jMKt9f93fZwJHVQoov0hFAFwJ.jpg",
      "credit_id": "5e2f8fdcac8e6b0015bd5c86",
      "department": "Writing",
      "job": "Screenplay"
    },


I am sorry that the API is not a great one! But this is still a though experiment on how shitty it could be to fix these cases!

The page looks so good overall ❤️

**Comment by systm-spec** [comment]
ready for next review

**Review by ReallyBoard123 — APPROVED** [review]

**Pull request merged**